docs/topics/request-response.rst

```diff
@@ -312,6 +312,143 @@ errors if needed::
+:class:`~scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware`), where you
+need the ability to generate a short, unique identifier from a
+:class:`~scrapy.http.Request` object: a request fingerprint.
+
+You often do not need to worry about request fingerprints, the default request
+fingerprinter works for most projects.
+
+However, there is no universal way to generate a unique identifier from a
+request, because different situations require comparing requests differently.
+For example, sometimes you may need to compare URLs case-insensitively, include
+URL fragments, exclude certain URL query parameters, include some or all
+headers, etc.
+
+To change how request fingerprints are built for your requests, use the
+:setting:`REQUEST_FINGERPRINTER_CLASS` setting.
+
+.. setting:: REQUEST_FINGERPRINTER_CLASS
+
+REQUEST_FINGERPRINTER_CLASS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Default: ``'scrapy.utils.request.RequestFingerprinter'``
+
+The import path of a :ref:`request fingerprinter
+<custom-request-fingerprinter>`.
+
+The default request fingerprinter takes into account a canonical version
+(:func:`w3lib.url.canonicalize_url`) of
+:attr:`request.url <scrapy.http.Request.url>` and the values of
+:attr:`request.method <scrapy.http.Request.method>` and
+:attr:`request.body <scrapy.http.Request.body>`. It then generates an
+`SHA1 <https://en.wikipedia.org/wiki/SHA-1>`_ hash from their concatenation.
+
+
+.. _custom-request-fingerprinter:
+
+Writing your own request fingerprinter
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A request fingerprinter is a class that must implement the following method:
+
+.. method:: fingerprint(self, request)
+
+   Return a :class:`str` that uniquely identifies *request*.
+
+   The returned :class:`str` must be a valid filename. The hexadecimal
+   representation of a hash is recommended.
+
+   :param request: request to fingerprint
+   :type request: scrapy.http.Request
+
+Additionally, they may also implement the following methods:
+
+.. classmethod:: from_crawler(cls, crawler)
+
+   If present, this class method is called to create a request fingerprinter
+   instance from a :class:`~scrapy.crawler.Crawler` object. It must return a
+   new instance of the request fingerprinter.
+
+   *crawler* provides access to all Scrapy core components like settings and
+   signals; it is a way for the request fingerprinter to access them and hook
+   its functionality into Scrapy.
+
+   :param crawler: crawler that uses this pipeline
+   :type crawler: :class:`~scrapy.crawler.Crawler` object
+
+.. classmethod:: from_settings(cls, settings)
+
+   If present, and ``from_crawler`` is not defined, this class method is called
+   to create a request fingerprinter instance from a
+   :class:`~scrapy.settings.Settings` object. It must return a new instance of
+   the request fingerprinter.
+
+For some common use cases you can use
+:func:`~scrapy.utils.request.request_fingerprint` in your ``fingerprint``
+method implementation:
```

Comment:
Should we mention that `scrapy.utils.request.request_fingerprint` is used in the default implementation?